// Copyright 2022 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package google.cmrt.sdk.parameter_service.v1;

option java_multiple_files = true;
option java_outer_classname = "ParameterServiceProto";
option java_package = "com.google.cmrt.sdk.parameter_service.v1";

import "cc/core/common/proto/common.proto";

// For faster allocations of sub-messages.
option cc_enable_arenas = true;

// Fetches stored parameters from cloud. In AWS, they are stored in
// ParameterStore. In GCP, they are stored in SecretManager.
service ParameterService {
  // Fetches parameter from cloud.
  rpc GetParameter(GetParameterRequest) returns (GetParameterResponse) {}
}

// Request to get parameter.
message GetParameterRequest {
  // The given name of the parameter to be retrieved.
  string parameter_name = 1;
}

// Response of getting parameter.
message GetParameterResponse {
  // The execution result.
  scp.core.common.proto.ExecutionResult result = 1;
  // Returned parameter value.
  string parameter_value = 2;
}
